### PR TITLE
upgrade to org.locationtech.jts

### LINKED ISSUE #17: Upgrade to org.locationtech.jts
The library com.vividsolutions.jts was moved to org.locationtech.jts. The new maven dependency looks like this: 

```
<dependency>
    <groupId>org.locationtech.jts</groupId>
    <artifactId>jts</artifactId>
    <version>1.15.0</version>
</dependency>
```

Unfortunately, due to fact that the package name was changed, it is not a drop-in replacement and requires code changes to jackson-datatype-jts. This in turn would be a non-backwards-compatible change to jackson-datatype-jts, which might require a new major version?

If desired, I could provide a PR for these changes.

### PR DESCRIPTION
**Pull request by karussell**
Fixes #17

### CODE REVIEW

**Comment by Sarev0k** [comment]
@akbertram @purpleP Would it be possible for you to merge this and kick off a new release?

I'd like to consume the latest version of JTS, and this is blocking me.

**Comment by karussell** [comment]
See https://github.com/graphhopper/jackson-datatype-jts/ and also [here](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.graphhopper.external%22%20AND%20a%3A%22jackson-datatype-jts%22) for a bug fix release.

**Comment by ernstdevreede** [comment]
Can we expect this very useful PR for switching to org.locationtech.jts package to be merged soon?
Otherwise I might want to clone the project.

**Comment by russellhoff** [comment]
Really interesting PR. Please go on! Thank you so much.

**Comment by roryodonnell** [comment]
I can confirm this issue as I'm also experiencing it
```
[INFO] +- com.bedatadriven:jackson-datatype-jts:jar:2.4:compile
[INFO] |  \- com.vividsolutions:jts:jar:1.13:compile
```
which causes stack overflow (infinite recursion) when serializing an object with `org.locationtech.jts.geom.*` object members

File: pom.xml [hunk 1]
```diff
@@ -62,9 +62,9 @@
       <version>2.4.2</version>
```
Comment (herekkratz) [inline pom.xml:62]: The incompatibility justifies a major version increment.
```suggestion
      <version>2.5.0</version>
```

File: pom.xml [hunk 2]
```diff
@@ -62,9 +62,9 @@
...
     </dependency>
     <dependency>
-      <groupId>com.vividsolutions</groupId>
-      <artifactId>jts</artifactId>
-      <version>1.13</version>
+      <groupId>org.locationtech.jts</groupId>
+      <artifactId>jts-core</artifactId>
+      <version>1.15.0</version>
```
Comment (herekkratz) [inline pom.xml:67]:
```suggestion
      <version>1.16.0</version>
```

**Review by herekkratz — CHANGES_REQUESTED** [review]
Looks good to me, except for the two proposed changes.

**Comment by russellhoff** [comment]
How can I lend a hand?

**Comment by BartoszKaszewczuk** [comment]
Any reason why this wasn't merged yet?

**Comment by purpleP** [comment]
Guys, I don't have motivation to work on this and even if I would I don't think I can upload new releases to maven central etc.

If I would maintain this library I would do this

- use gradle as a build tool
- use property based testing instead of or along with existing tests.
- upgrade to java 8 or maybe even 9.

**Review by larsnaesbye — APPROVED** [review]

**Pull request closed without merging**